### PR TITLE
Add method 'light/get_balances'

### LINKED ISSUE #120: Proposal: implement a method in WS API to get address balances
In the current WS API there is no easy way to get an account balances. There is the method `light/get_history` which give in and out transactions of a specific address(es) but it's not that easy to compute the balances giving the possibilities that there could be smart contract, asset created or it could simply not work if there is more transactions than the `MAX_HISTORY_ITEMS` limit. It would be great to have directly the balance from a new API method without having to compute it yourself from history.

### PR DESCRIPTION
**Pull request by bonustrack**
Fix #120

### CODE REVIEW

File: network.js [hunk 1]
```diff
@@ -2706,6 +2706,50 @@ function handleRequest(ws, tag, command, params){
...
+				return sendErrorResponse(ws, tag, "addresses must be non-empty array");
+			if (!addresses.every(ValidationUtils.isValidAddress))
+				return sendErrorResponse(ws, tag, "some addresses are not valid");
+			if (addresses.length > 100)
+				return sendErrorResponse(ws, tag, "too many addresses");
+			async.map(addresses, function(address, cb) {
+				db.query(
+					"SELECT asset, is_stable, SUM(amount) AS balance \n\
+					FROM outputs JOIN units USING(unit) \n\
+					WHERE is_spent=0 AND address=? AND sequence='good' \n\
```
Comment (tonyofbyteball) [inline network.js]: with `address IN(?)` and `GROUP BY address` it can be done in one query and should run faster

**Comment by bonustrack** [comment]
This PR is ready for another review @tonyofbyteball

**Comment by tonyofbyteball** [comment]
Thanks, a few comments:
- it seems there are conflicts
- please don't use `const` as this code has to run on android 4.4
- it seems the clients have to rely on the order of addresses in the returned array being the same as in the request.  I think it would be easier to interpret the result if the result is already indexed by `address`: balances[address][asset]={stable:123, pending:456}.  It would also obviate the need to iterate through all rows for each address.

**Comment by bonustrack** [comment]
Thx @tonyofbyteball i've made the changes you asked, i've simplified the code but now the addresses with empty balance will not be visible, would that work for you?

**Comment by tonyofbyteball** [comment]
Yes that's ok.

**Pull request merged**